syntax = "proto3";
import "google/protobuf/timestamp.proto";
import "google/protobuf/empty.proto";

message AddTodoRequest{

    string title = 1;
    string details = 2;
    google.protobuf.Timestamp deadline = 3;

}

message AddTodoResponse{

    string uuid = 1;
    string title = 2;
    string details = 3;
    google.protobuf.Timestamp deadline = 4;

}

message DeleteTodoRequest{
    string uuid = 1;
}

message DeleteTodoResponse{
    string status = 1;
}

message GetTodosRequest{
    google.protobuf.Empty null = 1;
}


message GetTodosResponse{
    repeated AddTodoResponse todos = 1;

}


service Todo{
    rpc AddTodo(AddTodoRequest) returns (AddTodoResponse) {}
    rpc DeleteTodo(DeleteTodoRequest) returns (DeleteTodoResponse){}
    rpc GetTodos(GetTodosRequest) returns (GetTodosResponse) {}

}

